contracts/mocks/MockValidator.sol

```diff
@@ -9,27 +9,35 @@ import { ECDSA } from "solady/src/utils/ECDSA.sol";
 import { SignatureCheckerLib } from "solady/src/utils/SignatureCheckerLib.sol";
 import { MessageHashUtils } from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
 import { ERC7739Validator } from "../base/ERC7739Validator.sol";
-import { IERC1271Legacy } from "../interfaces/modules/IERC1271Legacy.sol";
 
-contract MockValidator is ERC7739Validator, IERC1271Legacy {
+contract MockValidator is ERC7739Validator {
     mapping(address => address) public smartAccountOwners;
 
     function validateUserOp(PackedUserOperation calldata userOp, bytes32 userOpHash) external view returns (uint256 validation) {
         address owner = smartAccountOwners[msg.sender];
         return _validateSignatureForOwner(owner, userOpHash, userOp.signature) ? VALIDATION_SUCCESS : VALIDATION_FAILED;
     }
 
-    function isValidSignatureWithSender(address, bytes32 hash, bytes calldata signature) external view returns (bytes4) {
-        address owner = smartAccountOwners[msg.sender];
-
-        (bytes32 computeHash, bytes calldata truncatedSignature) = _erc1271HashForIsValidSignatureViaNestedEIP712(hash, signature);
-
-        return _validateSignatureForOwner(owner, computeHash, truncatedSignature) ? ERC1271_MAGICVALUE : ERC1271_INVALID;
-    }
-
-    function isValidSignatureWithSenderLegacy(address, bytes32 hash, bytes calldata signature) external view returns (bytes4) {
-        address owner = smartAccountOwners[msg.sender];
-        return _validateSignatureForOwner(owner, hash, signature) ? ERC1271_MAGICVALUE : ERC1271_INVALID;
+function isValidSignatureWithSender(
```

Comment:
correct me if I'm wrong. in this branch now MockValidator is non-ERC7739 validator and K1Validator is ERC7739 and both can validate signatures based on the passed hash. and still protects from cases like more accounts shared by common 1271 signers